app/bundles/FormBundle/Model/FormModel.php

```diff
@@ -729,6 +731,75 @@ public function populateValuesWithGetParameters(Form $form, &$formHtml): void
         }
     }
 
+    /**
+     * Replaces strings the PHP date string parser understands with
+     * the actual time in date, datetime and hidden fields.
+     *
+     * Hidden fields must end with `|date` to prevent replacing
+     * strings by a date time representation by accident.
+     */
+    public function populateDateTimeValues(Form $form, string &$formHtml): void
+    {
+        $formName       = $form->generateFormName();
+        $fields         = $form->getFields();
+        $autoFillFields = [];
+
+        /** @var Field $field */
+        foreach ($fields as $key => $field) {
+            if (in_array($field->getType(), ['date', 'datetime', 'hidden'])) {
+                $autoFillFields[$key] = $field;
+            }
+        }
+
+        foreach ($autoFillFields as $field) {
+            $fieldType = $field->getType();
+            $token     = $field->getDefaultValue();
+
+            switch ($fieldType) {
+                case 'date':
+                    $format = 'Y-m-d';
```

Comment:
Just to note, there is a global setting for date formats. But the default values may be too difficult for form inputs:
![Screenshot 2024-09-09 at 12 51 16](https://github.com/user-attachments/assets/6fe27192-5934-4278-aef6-600a80d2e5ad)
So no need to implement this. Just wanted to note there is an option to avoid hard-coded formats.

Reply:
You're right, but date/time HTML input fields rely on a very narrow input format. The hidden field will most likely get parsed somewhere, too. E.g., in a custom lead field. Thus, hard coding it to the ISO 8601 format made sense to me.

Reply:
I agree